src/Compilers/CSharp/Portable/Symbols/NamespaceOrTypeSymbol.cs

```diff
@@ -317,9 +317,11 @@ public virtual ImmutableArray<NamedTypeSymbol> GetTypeMembers(string name, int a
 
             foreach (var named in namespaceOrTypeMembers)
             {
-                if (!named.MangleName && (forcedArity == -1 || forcedArity == named.Arity))
+                if (!named.MangleName &&
+                    (forcedArity == -1 || forcedArity == named.Arity) &&
+                    named.MetadataName == emittedTypeName.TypeName)
```

Comment:
>named.MetadataName == emittedTypeName.TypeName [](http://example.com/codeflow?start=20&length=46)

Is there a test backing this change?

Reply:
Yes. This part is tested where I added an assert of `C` being null. Since `C` is not a mangled name algorithm of its search follows this code path